source/errors.ts

```diff
@@ -1,11 +1,38 @@
-'use strict';
-const urlLib = require('url');
-const http = require('http');
-const PCancelable = require('p-cancelable');
-const is = require('@sindresorhus/is');
-
-class GotError extends Error {
-	constructor(message, error, options) {
+import urlLib from 'url';
+import http from 'http';
+import PCancelable from 'p-cancelable';
+import is from '@sindresorhus/is';
+
+// TODO: Use the Got options-object type
+interface Options {
+	host: string;
+	hostname: string;
+	method: string;
+	path: string;
+	socketPath: string;
+	protocol: string;
+	href: string;
+	options: string;
+}
+
+export class GotError extends Error {
+	code: number | undefined;
+
+	body: string | undefined;
+
+	statusCode: number | undefined;
+
+	statusMessage: string | undefined;
+
+	headers: string | undefined;
+
+	redirectUrls: string | undefined;
+
+	event: string | undefined;
+
+	timings: number | undefined;
```

Comment:
Some of the properties should be moved to different errors, for example, `timings` should be moved into `TimeoutError`.